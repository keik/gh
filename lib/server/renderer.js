// @flow

/**
 * Return full HTML string with React props
 *
 * @param {string} content content as HTML string
 * @param {object} props React props object for Server-Side Rendering
 * @param {string} entryName name of entrypoint
 */
export function renderFullPage(
  content: string,
  props: any,
  entryName: string,
  bust: number | string
) {
  return `
 <!DOCTYPE html>
 <html>
   <head>
     <meta charset="utf-8">
     <title>gh</title>
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.6.3/css/font-awesome.min.css">
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.7.0/styles/github.min.css">
     <link href="https://fonts.googleapis.com/css?family=Rajdhani" rel="stylesheet" />
     <link rel="stylesheet" href="/style.css?${bust}">
   </head>
   <body>
     <div id="app">${content}</div>
     <script>APP_PROPS = ${JSON.stringify(props)}</script>
     <script src="/${entryName}.js?${bust}"></script>
   </body>
 </html>
 `
}
