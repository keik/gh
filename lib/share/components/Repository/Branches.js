// @flow

import * as React from 'react'
import { connect } from 'react-redux'
import type { Dispatch } from 'redux'

import Panel from '../common/blocks/Panel'
import List from '../common/layouts/List'
import * as BranchesAction from '../../ducks/repository/branches'
import type { BranchT } from '../../../types/gh'

export const Branches = ({
  branches,
  defaultBranchName,
  params
}: {
  branches: Array<BranchT>,
  defaultBranchName: string,
  params: { owner: string, repo: string }
}) => {
  const defaultBranch: BranchT = branches.find(
    branch => branch.name === defaultBranchName
  ) || {
    commit: { author: { date: '', name: '' }, message: '', sha: '' },
    name: ''
  }
  const activeBranches = branches.filter(
    branch => branch.name !== defaultBranchName
  )
  return (
    <div>
      <Panel>
        <Panel.Header>Default branch</Panel.Header>
        <Panel.Body>
          <List lined>
            <li>
              <a
                href={`/${params.owner}/${params.repo}/tree/${
                  defaultBranch.name
                }`}
              >
                {defaultBranch.name}
              </a>
              &nbsp; Updated{' '}
              <time-ago datetime={defaultBranch.commit.author.date} /> by&nbsp;
              <a href={`/${defaultBranch.commit.author.name}`}>
                {defaultBranch.commit.author.name}
              </a>
            </li>
          </List>
        </Panel.Body>
      </Panel>
      <Panel>
        <Panel.Header>Active branches</Panel.Header>
        <Panel.Body>
          <List lined>
            {activeBranches.map((branch, i) => (
              <li key={i}>
                <a href={`/${params.owner}/${params.repo}/tree/${branch.name}`}>
                  {branch.name}
                </a>
                &nbsp; Updated <time-ago datetime={branch.commit.author.date} />{' '}
                by&nbsp;
                <a href={`/${branch.commit.author.name}`}>
                  {branch.commit.author.name}
                </a>
              </li>
            ))}
          </List>
        </Panel.Body>
      </Panel>
    </div>
  )
}

export default connect<_, _, *, _, *, _>(({ branches }) => ({
  branches
}))(
  class BranchesContainer extends React.Component<{
    branches: Array<BranchT>,
    defaultBranchName: string,
    dispatch: Dispatch<*>,
    params: { owner: string, repo: string }
  }> {
    async componentDidMount() {
      const {
        dispatch,
        params: { owner, repo }
      } = this.props
      dispatch(await BranchesAction.fetch({ owner, repo }))
    }

    render() {
      return <Branches {...this.props} />
    }
  }
)
