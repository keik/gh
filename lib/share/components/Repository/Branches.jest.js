// @flow

import * as React from 'react'
import renderer from 'react-test-renderer'

import { Branches } from './Branches'

test('<Branches /> snapshot', () => {
  const tree = renderer
    .create(
      <Branches
        branches={[
          {
            commit: {
              author: {
                date: '$DATE',
                name: '$AUTHOR'
              },
              message: '$MESSAGE',
              sha: '$SHA'
            },
            name: '$BRANCH'
          }
        ]}
        defaultBranchName="master"
        params={{ owner: '$OWNER', repo: '$REPO' }}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
