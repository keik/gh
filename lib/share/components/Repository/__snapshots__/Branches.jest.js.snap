// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Branches /> snapshot 1`] = `
<div
  className="InnerContainer-sc-1toru5t-0 eVVvlf"
>
  <div
    className="Panel-sc-1hnzj4y-0 ccxQvz"
  >
    <div
      className="Panel__Header-sc-1hnzj4y-1 chmgYi"
    >
      Default branch
    </div>
    <div
      className="Panel__Body-sc-1hnzj4y-2 dwhEKI"
    >
      <ul
        className="Branches__branches___39ZZS"
      >
        <li>
          <a
            href="/$OWNER/$REPO/tree/"
          >
            
          </a>
            Updated
           
          <time-ago
            datetime=""
          />
           by 
          <a
            href="/"
          >
            
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="Panel-sc-1hnzj4y-0 ccxQvz"
  >
    <div
      className="Panel__Header-sc-1hnzj4y-1 chmgYi"
    >
      Active branches
    </div>
    <div
      className="Panel__Body-sc-1hnzj4y-2 dwhEKI"
    >
      <ul
        className="Branches__branches___39ZZS"
      >
        <li>
          <a
            href="/$OWNER/$REPO/tree/$BRANCH"
          >
            $BRANCH
          </a>
            Updated 
          <time-ago
            datetime="$DATE"
          />
           
          by 
          <a
            href="/$AUTHOR"
          >
            $AUTHOR
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
